Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -471,34 +472,76 @@ public class PBXProjGenerator {
             .filter { $0.extension == "lproj" }
             .sorted { $0.lastComponent < $1.lastComponent }
 
-        var groupChildren: [String] = []
-        var allFilePaths: [Path] = filePaths
+        var groupChildren: [String] = filePaths.map { getFileReference(path: $0, inPath: path) }
+        var allSourceFiles: [SourceFile] = filePaths.map { generateSourceFile(path: $0) }
         var groups: [PBXGroup] = []
 
         for path in directories {
             let subGroups = try getGroups(path: path, depth: depth + 1)
-            allFilePaths += subGroups.filePaths
+            allSourceFiles += subGroups.sourceFiles
             groupChildren.append(subGroups.groups.first!.reference)
             groups += subGroups.groups
         }
 
-        for filePath in filePaths {
-            let fileReference = getFileReference(path: filePath, inPath: path)
-            groupChildren.append(fileReference)
+        // create variant groups of the base localisation first
+        var baseLocalisationVariantGroups:[PBXVariantGroup] = []
+        if let baseLocalisedDirectory = localisedDirectories.first(where: { $0.lastComponent == "Base.lproj" }) {
+            for path in try baseLocalisedDirectory.children() {
+                let filePath = "\(baseLocalisedDirectory.lastComponent)/\(path.lastComponent)"
+
+                let variantGroup: PBXVariantGroup
+                if let cachedGroup = variantGroupsByPath[path] {
+                    variantGroup = cachedGroup
+                } else {
+                    variantGroup = PBXVariantGroup(reference: generateUUID(PBXVariantGroup.self, filePath),
+                                                   children: [],
+                                                   name: path.lastComponent,
+                                                   sourceTree: .group)
+                    variantGroupsByPath[path] = variantGroup
+                    
+                    addObject(variantGroup)
+                    groupChildren.append(variantGroup.reference)
+                }
+
+                baseLocalisationVariantGroups.append(variantGroup)
+
+                let buildFile = PBXBuildFile(reference: generateUUID(PBXBuildFile.self, variantGroup.reference), fileRef: variantGroup.reference, settings: nil)
+                allSourceFiles.append(SourceFile(path: path, fileReference: variantGroup.reference, buildFile: buildFile))
+            }
         }
 
+        // add references to localised resources into base localisation variant groups
         for localisedDirectory in localisedDirectories {
+            let localisationName = localisedDirectory.lastComponentWithoutExtension
             for path in try localisedDirectory.children().sorted { $0.lastComponent < $1.lastComponent } {
+                guard fileReferencesByPath[path] == nil else {
```

Comment:
What happens when multiple targets share localised sources that aren't in a `Base.lproj` directory? It looks like the second target won't get a build file, as this path is skipped

Reply:
You’re right. I will fix it.

Reply:
@yonaskolb I fixed this. https://github.com/yonaskolb/XcodeGen/pull/70/commits/859609158cbbe94930962c9c13848e3e7e46232c#diff-2f3c621620623f55e7ff42cc32198a10R540